pkg/attacher/lister.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package attacher
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+
+	"google.golang.org/grpc"
+)
+
+// Lister implements list operations against a remote CSI driver.
+type VolumeLister interface {
+	// ListVolumes calls ListVolumes on the driver and returns a map with keys
+	// of VolumeID and values of the list of Node IDs that volume is published
+	// on
+	ListVolumes(ctx context.Context) (map[string][]string, error)
```

Comment:
optional: if order never matters, maybe this could return `map[string]sets.String`?

feel free to ignore if you don't want to add a dep on `sets`

Reply:
I remember reading that it is more efficient in golang to iterate over "small lists" to find an item than look it up in a map. I don't expect the list of nodes that a volume is attached to to ever get large (but this can always be fixed if that assumption changes in the future)

Reply:
fair, I was thinking primarily about readability, but a small loop isn't that bad.